### PR TITLE
PipelineHandler implementation

### PR DESCRIPTION
**Pull request by jpopelka**
I'm playing with this on staging, so there might still be some fixups/amends, but mostly it should be ready for review.

Prerequisite packit/packit-service#1229

### CODE REVIEW

**Comment by jpopelka** [comment]
recheck

File: README.md [hunk 1]
```diff
@@ -11,9 +11,9 @@ but is expected to eventually follow [this workflow](https://github.com/packit/r
 
 ##### Must have:
 
-- [ ] If a user creates a merge-request on the source-git repository:
+- [x] If a user creates a merge-request on the source-git repository:
   - [x] Create a matching merge-request to the dist-git repository.
-  - [ ] Sync the CI results from the dist-git merge-request to the source-git merge-request.
+  - [x] Sync the CI results from the dist-git merge-request to the source-git merge-request.
```
Comment (TomasTomecek) [inline README.md:16]: \o/

File: hardly/handlers/distgit.py [hunk 2]
```diff
@@ -99,35 +107,115 @@ def run(self) -> TaskResults:
...
+
+    @property
+    def status_reporter(self) -> StatusReporter:
+        if not self._status_reporter:
+            self._status_reporter = StatusReporter.get_instance(
+                project=self.src_git_project,
+                # The head_commit is latest commit of the MR.
+                # If there was a new commit pushed before the pipeline ended, the report
+                # might be incorrect until the new (for the new commit) pipeline finishes.
+                commit_sha=self.src_git_mr.head_commit,
```
Comment (TomasTomecek) [inline hardly/handlers/distgit.py:187]: can we store the commit info in our database so we don't hit this race condition?

Comment (TomasTomecek) [reply hardly/handlers/distgit.py:187]: as a fallowup work

**Pull request merged**